File: src/Application Specific/Classes/Entity Collections/Spares/PartSupplier/PartSupplier.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace PartSuppliers

        Public Class PartSupplier

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "PartSupplier Properties"


            Private _PartSupplierID As Integer = 0
            Public ReadOnly Property PartSupplierID() As Integer
                Get
                    Return _PartSupplierID
                End Get
            End Property
            Public WriteOnly Property SetPartSupplierID() As Object
                Set(Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PartSupplierID", Value)
                End Set
            End Property


            Private _PartCode As String = String.Empty
            Public ReadOnly Property PartCode() As String
                Get
                    Return _PartCode
                End Get
            End Property
            Public WriteOnly Property SetPartCode() As Object
                Set(Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PartCode", Value)
                End Set
            End Property


            Private _PartID As Integer = 0
            Public ReadOnly Property PartID() As Integer
                Get
                    Return _PartID
                End Get
            End Property
            Public WriteOnly Property SetPartID() As Object
                Set(Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PartID", Value)
                End Set
            End Property


            Private _SupplierID As Integer = 0
            Public ReadOnly Property SupplierID() As Integer
                Get
                    Return _SupplierID
                End Get
            End Property
            Public WriteOnly Property SetSupplierID() As Object
                Set(Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SupplierID", Value)
                End Set
            End Property

            Private _Price As Double = 0
            Public ReadOnly Property Price() As Double
                Get
                    Return _Price
                End Get
            End Property
            Public WriteOnly Property SetPrice() As Object
                Set(Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Price", Value)
                End Set
            End Property

            Private _SecondaryPrice As Double = 0
            Public ReadOnly Property SecondaryPrice() As Double
                Get
                    Return _SecondaryPrice
                End Get
            End Property
            Public WriteOnly Property SetSecondaryPrice() As Object
                Set(Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SecondaryPrice", Value)
                End Set
            End Property

            Private _QuantityInPack As Double = 0
            Public ReadOnly Property QuantityInPack() As Double
                Get
                    Return _QuantityInPack
                End Get
            End Property
            Public WriteOnly Property SetQuantityInPack() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuantityInPack", Value)
                End Set
            End Property

            Private _preferred As Boolean = False
            Public ReadOnly Property Preferred() As Boolean
                Get
                    Return _preferred
                End Get
            End Property
            Public WriteOnly Property SetPreferred() As Boolean
                Set(ByVal Value As Boolean)
                    _preferred = Value
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Spares/PartSupplier/PartSupplier.cs
using System.Collections;

namespace FSM.Entity
{
    namespace PartSuppliers
    {
        public class PartSupplier
        {
            private DataTypeValidator _dataTypeValidator;

            public PartSupplier()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _PartSupplierID = 0;

            public int PartSupplierID
            {
                get
                {
                    return _PartSupplierID;
                }
            }

            public object SetPartSupplierID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PartSupplierID", value);
                }
            }

            private string _PartCode = string.Empty;

            public string PartCode
            {
                get
                {
                    return _PartCode;
                }
            }

            public object SetPartCode
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PartCode", value);
                }
            }

            private int _PartID = 0;

            public int PartID
            {
                get
                {
                    return _PartID;
                }
            }

            public object SetPartID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PartID", value);
                }
            }

            private int _SupplierID = 0;

            public int SupplierID
            {
                get
                {
                    return _SupplierID;
                }
            }

            public object SetSupplierID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SupplierID", value);
                }
            }

            private double _Price = 0;

            public double Price
            {
                get
                {
                    return _Price;
                }
            }

            public object SetPrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Price", value);
                }
            }

            private double _SecondaryPrice = 0;

            public double SecondaryPrice
            {
                get
                {
                    return _SecondaryPrice;
                }
            }

            public object SetSecondaryPrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SecondaryPrice", value);
                }
            }

            private double _QuantityInPack = 0;

            public double QuantityInPack
            {
                get
                {
                    return _QuantityInPack;
                }
            }

            public object SetQuantityInPack
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuantityInPack", value);
                }
            }

            private bool _preferred = false;

            public bool Preferred
            {
                get
                {
                    return _preferred;
                }
            }

            public bool SetPreferred
            {
                set
                {
                    _preferred = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}